src/axolotl/integrations/nemo_gym/server.py

```
@@ -0,0 +1,238 @@
+
+    # Check if already running
+    try:
+        requests.get(head_url, timeout=2)
+        LOG.info(f"NeMo Gym servers already running on port {head_port}.")
+        return
+    except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
+        pass
+
+    config_str = ",".join(config_paths)
+    _ng_log_file = open(os.path.join(gym_dir, "ng_run.log"), "w")  # noqa: SIM115
+    _ng_process = subprocess.Popen(  # nosec
```

Ruff (0.15.6)

[error] `subprocess` call: check for execution of untrusted input

(S603)